api/v1alpha1/serviceintentions_types_test.go

```diff
@@ -738,141 +808,36 @@ func TestServiceIntentions_Validate(t *testing.T) {
 					},
 				},
 			},
-			`serviceintentions.consul.hashicorp.com "does-not-matter" is invalid: spec.sources[0]: Invalid value: "{\"name\":\"svc-2\",\"namespace\":\"bar\",\"action\":\"deny\",\"permissions\":[{\"action\":\"allow\",\"http\":{\"pathExact\":\"/bar\"}}]}": action and permissions are mutually exclusive and only one of them can be specified`,
-		},
-	}
-	for name, testCase := range cases {
-		t.Run(name, func(t *testing.T) {
-			err := testCase.input.Validate()
-			if testCase.expectedErrMsg != "" {
-				require.EqualError(t, err, testCase.expectedErrMsg)
-			} else {
-				require.NoError(t, err)
-			}
-		})
-	}
-}
-
-func TestServiceIntentions_ValidateNamespaces(t *testing.T) {
-	cases := map[string]struct {
-		namespacesEnabled bool
-		input             *ServiceIntentions
-		expectedErrMsg    string
-	}{
-		"enabled: valid": {
-			true,
-			&ServiceIntentions{
-				ObjectMeta: metav1.ObjectMeta{
-					Name: "does-not-matter",
-				},
-				Spec: ServiceIntentionsSpec{
-					Destination: Destination{
-						Name:      "dest-service",
-						Namespace: "namespace",
-					},
-					Sources: SourceIntentions{
-						{
-							Name:      "web",
-							Namespace: "web",
-							Action:    "allow",
-						},
-						{
-							Name:      "db",
-							Namespace: "db",
-							Action:    "deny",
-						},
-						{
-							Name:      "bar",
-							Namespace: "bar",
-							Permissions: IntentionPermissions{
-								{
-									Action: "allow",
-									HTTP: &IntentionHTTPPermission{
-										PathExact:  "/foo",
-										PathPrefix: "/bar",
-										PathRegex:  "/baz",
-										Header: IntentionHTTPHeaderPermissions{
-											{
-												Name:    "header",
-												Present: true,
-												Exact:   "exact",
-												Prefix:  "prefix",
-												Suffix:  "suffix",
-												Regex:   "regex",
-												Invert:  true,
-											},
-										},
-										Methods: []string{
-											"GET",
-											"PUT",
-										},
-									},
-								},
-							},
-							Description: "an L7 config",
-						},
-					},
-				},
+			namespacesEnabled: true,
+			expectedErrMsgs: []string{
+				`serviceintentions.consul.hashicorp.com "does-not-matter" is invalid: spec.sources[0]: Invalid value: "{\"name\":\"svc-2\",\"namespace\":\"bar\",\"action\":\"deny\",\"permissions\":[{\"action\":\"allow\",\"http\":{\"pathExact\":\"/bar\"}}]}": action and permissions are mutually exclusive and only one of them can be specified`,
 			},
-			"",
 		},
-		"disabled: destination namespace specified": {
-			false,
-			&ServiceIntentions{
+		"namespaces disabled: destination namespace specified": {
+			input: &ServiceIntentions{
 				ObjectMeta: metav1.ObjectMeta{
 					Name: "does-not-matter",
 				},
 				Spec: ServiceIntentionsSpec{
 					Destination: Destination{
 						Name:      "dest-service",
-						Namespace: "foo",
+						Namespace: "namespaceA",
```

Comment:
minor nit here, but I ran into this problem the other day when building a test that had an upper case letter for Namespaces in consul. Consul will automatically convert this to all lowercase letters, so if you're asserting later that the output is `namespaceA` (which I don't believe you are) *from Consul* it will not pass. I think it's best to keep our tests with lowercase namespaces as those are valid for both k8s and Consul.
What do you think?

```suggestion
						Namespace: "namespace-a",
```

Reply:
>I think it's best to keep our tests with lowercase namespaces as those are valid for both k8s and Consul.

Yeah, I agree, I'll make this change!

Are the namespaces in consul tied to the corresponding k8s namespace? Would anyone ever write a serviceintention for example with a namespace with capital letters? Or because the namespace is inherently linked between consul and k8s it will only ever be lowercase in both?

Reply:
Actually capital letters are not even valid in k8s, so it would cover both!
```
kyle@kyles-mbp consul-helm % kubectl create namespace ABCD
The Namespace "ABCD" is invalid: metadata.name: Invalid value: "ABCD": a DNS-1123 label must consist of lower case alphanumeric characters or '-', and must start and end with an alphanumeric character (e.g. 'my-name',  or '123-abc', regex used for validation is '[a-z0-9]([-a-z0-9]*[a-z0-9])?')
```

Reply:
Got it, thanks for the explanation!

Reply:
Yeah great catch!